testing/integration_testing/restore_secrets.py

```diff
@@ -0,0 +1,161 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Script for restoring secrets into the integration test projects.
+
+Usage:
+
+python restore_secrets.py --passphrase <passphrase> --repo_dir <path_to_repo>
+
+repo_dir refers to the C++ SDK github repository. Defaults to current directory.
+
+As an alternative to passing the passphrase as a flag, the password can be
+stored in a file and passed via the --passphrase_file flag.
+
+This will perform the following:
+
+- Google Service files (plist and json) will be restored into the
+  integration_test directories.
+- The server key will be patched into the Messaging project.
+- The uri prefix will be patched into the Dynamic Links project.
+- The reverse id will be patched into all Info.plist files, using the value from
+  the decrypted Google Service plist files as the source of truth.
+
+"""
+
+import argparse
```

Comment:
Let's switch this to absl.flags in expectation of using that for all python scripts.